app/scripts/modules/netflix/fastProperties/view/fastPropertyPods.component.ts

```diff
@@ -0,0 +1,76 @@
+import {module, isArray, IComponentController, IComponentOptions} from 'angular';
+import {Property} from '../domain/property.domain';
+import { flatten, values } from 'lodash';
+import {IApplicationStateParams} from '../../../core/application/application.state.provider';
+
+export class FastPropertyPodsController implements IComponentController {
+
+  public properties: any[];
+
+  static get $inject() {
+    return [
+      '$state',
+      '$stateParams'
+    ];
+  }
+
+  public $onInit() {
+    this.resetPropertyDetails();
+  }
+
+  constructor(
+    private $state: any,
+    public $stateParams: IApplicationStateParams
+  ) {}
+
+
+  public isPropertyListArray(): boolean {
+    return isArray(this.properties);
+  }
+
+  public isGrouped(): boolean {
+    return !this.isPropertyListArray();
+  }
+
+
+  private isDetailInPropInList() {
+    let included = false;
+
+    if (this.isPropertyListArray()) {
+      included = this.properties.some(this.hasPropId);
+    } else {
+      included = flatten(values(this.properties)).some(this.hasPropId);
+    }
+    return included;
+  };
+
+  hasPropId = (prop: Property) => {
+    return prop.propertyId === this.$stateParams['propertyId'];
+  }
+
+  resetPropertyDetails = () => {
+    if (this.$stateParams['propertyId'] && !this.isDetailInPropInList()) {
```

Comment:
should be able to omit the brackets and just call `this.$stateParams.propertyId`, thanks TS 2.2